app/templates/_widget.ts

```diff
@@ -0,0 +1,42 @@
+/// <reference path="../../../tsd.ts.ts" />
+/// <amd-dependency path="dojo/text!./templates/<%=widgetName%>.html" name="template" />
+declare var template: any;
+
+//dojo
+import dojoDeclare = require("dojo/_base/declare");
+
+//dijit
+import WidgetBase = require("dijit/_WidgetBase");
+import TemplatedMixin = require("dijit/_TemplatedMixin"); 
+<% if(widgetsInTemplate) { %>import WidgetsInTemplateMixin = require("dijit/_WidgetsInTemplateMixin"); <% }%>
+
+interface WidgetInterface {
```

Comment:
Hey Tom, sorry for the massive delay. Travel back from the UK then a tonne of work. Consulting it comes in waves!

Yes I'm only using the official dojo type definitions. I don't think dojoDelcare is ever going to bring back the type of the mixed in class because it would have to dynamically generate that type based on what you passed into dojo declare. I admit to not knowing some of the deeper functionality of ts but I think this is beyond the realms of what it can do.

I can add all of the relevant lifecycle events to the interface and the class depending on whether it's templated or not. TBH though I tend to just use startup and destroy for simplicities sake (occasionally postCreate). I'd hope people using widget would know the basic lifecycle events...

Not sure where to go from here, we either drop it and I just use my own branch. Or we see if we can get someone like Rene to comment.